File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        result = [-1] * k

        frequencies = {}
        for i in nums:
            frequencies[i] = 1 + frequencies.get(i, 0)

        i = 0
        n = 0
        greatest = 0
        for key in frequencies:
            if frequencies[key] > greatest:
                result[i] = key
                i = (i + 1) % k
        
        print(result)

        return result

File: Data Structures & Algorithms/counting-bits/submission-3.java
class Solution {
    public int[] countBits(int n) {
        int[] result = new int[n + 1];

        for (int i = 0; i <= n; i++) {
            int j = i;
            while (j != 0 && ++result[i] > 0)
                j = (j - 1) & j;

            // for (int j = i; j != 0 && ++result[i] > 0; j = (j - 1) & j);
        }

        return result;
    }
}
